libnetwork/drivers/overlay/ov_network.go

```diff
@@ -76,6 +76,14 @@ type network struct {
 
 func init() {
 	reexec.Register("set-default-vlan", setDefaultVlan)
+
+	// Lock main() to the initial thread to exclude the goroutines executing
+	// func (*network).watchMiss() from being scheduled onto that thread.
+	// Changes to the network namespace of the initial thread alter
+	// /proc/self/ns/net, which would break any code which (incorrectly)
+	// assumes that that file is a handle to the network namespace for the
+	// thread it is currently executing on.
+	runtime.LockOSThread()
```

Comment:
This doesn't seem right.
I get what it is doing here, but I think doing it in a `func init()` is not desirable.
I'm also not sure that it's strictly necessary to lock the main goroutine to its thread, but if we want to do that it should go in `func main()`

Reply:
> Why is it not ok to do it in package libnetwork? 

I have an aversion to `func init()` in library code because it forces things to happen for anyone who ever imports it, which I get is specifically what you kinda want here, but I just find it a bit heavy-handed.